Cython/Compiler/Symtab.py

```diff
@@ -1000,13 +1000,25 @@ def lookup_operator(self, operator, operands):
             method = obj_type.scope.lookup("operator%s" % operator)
             if method is not None:
                 arg_types = [arg.type for arg in operands[1:]]
-                res = PyrexTypes.best_match(arg_types, method.all_alternatives())
+                res = PyrexTypes.best_match(arg_types, method.all_alternatives(),
+                                            validate_types_fully=True)
                 if res is not None:
                     return res
         function = self.lookup("operator%s" % operator)
         function_alternatives = []
         if function is not None:
             function_alternatives = function.all_alternatives()
+        nonmember_alternatives = []
+        if self.is_cpp_class_scope and function and function.scope is self:
+            # for C++ classes we can have both member and non-member operators
+            # and we really want to consider both
+            outer_scope = self.outer_scope
+            while outer_scope and not outer_scope.is_module_scope:
+                outer_scope = outer_scope.module_scope
```

Comment:
Shouldn't this just read `global_scope = self.global_scope()`? Without the loop?